[schemagen] fails when $paxos table exists.

Hey!

**First of all, Please excuse my shallow knowledge if I miss something or considered noobe.**

I am exploring both Go and ScyllaDB. I run a cluster of 3 node of `scylladb/scylla:2025.4` in docker.

I have a user table and want to insert a unique record by email.

This is how I did it


```go
func insertUniqueByEmail(ctx context.Context, session gocqlx.Session, user models.UsersStruct) (models.UsersStruct, error) {
	now := time.Now().UTC()

	if user.UserId == (gocql.UUID{}) {
		user.UserId = gocql.TimeUUID()
	}
	if user.CreatedAt.IsZero() {
		user.CreatedAt = now
	}
	user.UpdatedAt = now

	var existingEmail string
	var existingUserId gocql.UUID

	stmt, names := models.UsersByEmail.InsertBuilder().Unique().ToCql()
	applied, err := session.Query(stmt, names).WithContext(ctx).BindStruct(user).ScanCAS(&existingEmail, &existingUserId)

	if err != nil {
		return models.UsersStruct{}, fmt.Errorf("Failed to execute unique insert: %w", err)
	}

	if !applied {
		return models.UsersStruct{}, errors.New("Email already exists!")
	}

	iq := models.Users.InsertQuery(session).WithContext(ctx).BindStruct(user)
	if err := iq.Exec(); err != nil {

		dq := models.UsersByEmail.DeleteQuery(session).WithContext(ctx).BindStruct(user)
		_ = dq.Exec()
		return models.UsersStruct{}, fmt.Errorf("User insertion failed: %w", err)
	}

	return user, nil
}
```

It works fine. however, from what I understand, ScyllaDB create internal table for LWT `user_by_email$paxos`

<img width="694" height="440" alt="Image" src="https://github.com/user-attachments/assets/289506c1-5c93-4b00-9132-88869a90fead" />


Now the issue is when I do changes in Table schema and re-generate the code, it fails with

```bash
$ 2026/01/31 22:27:29 failed to generate schema: render template: template: template: keyspace.tmpl:16:26: executing "keyspace.tmpl" at <camelize>: error calling camelize: not allowed name users_by_email$paxos
make: *** [makefile:19: code-gen] Error 1
```

From what I saw, that there is a method called `camelize` here https://github.com/scylladb/gocqlx/pull/201/changes#diff-f02315fd0ffeeb72b434f832ff541eddaad1f74f859bbec1c5910ba6399aa4f3 that caused this issue.

Is that considered a bug or do I miss something ?

If I may to suggest that we have a flag to ignore tables. Something like

```bash
-ignoreTables="user_by_email$paxos"
```

or we detect the name "$paxos" and exclude it from the generation.

For sure you know better in regard.

--- a/cmd/schemagen/schemagen.go
+++ b/cmd/schemagen/schemagen.go
@@ -104,15 +104,23 @@ func renderTemplate(md *gocql.KeyspaceMetadata) ([]byte, error) {
 	}
 
 	// Then remove all tables, views, and indices if their names match the
-	// filter.
+	// filter. Also skip internal Paxos tables created by LWT.
 	ignoredNames := make(map[string]struct{})
 	for _, ignoredName := range strings.Split(*flagIgnoreNames, ",") {
+		if ignoredName == "" {
+			continue
+		}
 		ignoredNames[ignoredName] = struct{}{}
 	}
 	for name := range ignoredNames {
 		delete(md.Tables, name)
 		delete(md.Views, name)
 		delete(md.Indexes, name)
+	}
+	for name := range md.Tables {
+		if strings.HasSuffix(name, "$paxos") {
+			delete(md.Tables, name)
+		}
 	}
 
 	removeUnusedUserTypes(md)

--- a/cmd/schemagen/schemagen_test.go
+++ b/cmd/schemagen/schemagen_test.go
@@ -209,6 +209,52 @@ func TestRenderTemplateRetainsNestedUserTypes(t *testing.T) {
 	}
 }
 
+func TestRenderTemplateIgnoresPaxosTables(t *testing.T) {
+	pkgname := *flagPkgname
+	ignoreNames := *flagIgnoreNames
+	ignoreIndexes := *flagIgnoreIndexes
+	t.Cleanup(func() {
+		*flagPkgname = pkgname
+		*flagIgnoreNames = ignoreNames
+		*flagIgnoreIndexes = ignoreIndexes
+	})
+
+	*flagPkgname = "schemagentest"
+	*flagIgnoreNames = ""
+	*flagIgnoreIndexes = false
+
+	idColumn := &gocql.ColumnMetadata{Name: "id", Type: "uuid"}
+	emailColumn := &gocql.ColumnMetadata{Name: "email", Type: "text"}
+	paxosColumn := &gocql.ColumnMetadata{Name: "row_key", Type: "blob"}
+	b, err := renderTemplate(&gocql.KeyspaceMetadata{
+		Tables: map[string]*gocql.TableMetadata{
+			"users": {
+				Name:           "users",
+				Columns:        map[string]*gocql.ColumnMetadata{"id": idColumn, "email": emailColumn},
+				OrderedColumns: []string{"id", "email"},
+				PartitionKey:   []*gocql.ColumnMetadata{idColumn},
+			},
+			"users_by_email$paxos": {
+				Name:           "users_by_email$paxos",
+				Columns:        map[string]*gocql.ColumnMetadata{"row_key": paxosColumn},
+				OrderedColumns: []string{"row_key"},
+				PartitionKey:   []*gocql.ColumnMetadata{paxosColumn},
+			},
+		},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	source := string(b)
+	if !strings.Contains(source, "Users = table.New") {
+		t.Fatalf("generated source does not include user table:\n%s", source)
+	}
+	if strings.Contains(source, "paxos") {
+		t.Fatalf("generated source includes internal Paxos table:\n%s", source)
+	}
+}
+
 func TestRenderTemplateRejectsNonComparableMapKeys(t *testing.T) {
 	pkgname := *flagPkgname
 	ignoreNames := *flagIgnoreNames

